libr/util/list.c

```diff
@@ -72,36 +73,39 @@ R_API void r_list_purge (RList *list) {
 }
 
 /* free the list */
-R_API void r_list_free (RList *list) {
+R_API void r_list_free(RList *list) {
 	if (list) {
 		r_list_purge (list);
 		R_FREE (list);
 	}
 }
 
-R_API bool r_list_delete_data (RList *list, void *ptr) {
+R_API bool r_list_delete_data(RList *list, void *ptr) {
 	void *p;
 	RListIter *iter;
 	r_list_foreach (list, iter, p) {
 		if (ptr == p) {
-			r_list_delete (list, iter);
+			r_list_delete (list, iter); //r_list_delete decrements 1 the length
+			list->length -= 1;
```

Comment:
According to the doc in the line above and to the code it seems this `list->length -= 1` is not needed because already done inside `r_list_delete`, right?